packages/eas-cli/src/commands/update/configure.ts

```diff
@@ -94,25 +191,74 @@ export default class UpdateConfigure extends EasCommand {
       await installExpoUpdatesAsync(projectDir);
     }
 
-    const hasAndroidNativeProject =
-      (await resolveWorkflowAsync(projectDir, Platform.ANDROID)) === Workflow.GENERIC;
-    const hasIosNativeProject =
-      (await resolveWorkflowAsync(projectDir, Platform.IOS)) === Workflow.GENERIC;
-    const isBare = hasAndroidNativeProject || hasIosNativeProject;
+    const [androidWorkflow, iosWorkflow] = await Promise.all([
+      resolveWorkflowAsync(projectDir, Platform.ANDROID),
+      resolveWorkflowAsync(projectDir, Platform.IOS),
+    ]);
 
-    await configureProjectForEASUpdateAsync(projectDir, exp, isBare);
+    const defaultRuntimeVersions = {
+      android:
+        androidWorkflow === Workflow.GENERIC
+          ? DEFAULT_BARE_RUNTIME_VERSION
+          : DEFAULT_MANAGED_RUNTIME_VERSION,
+      ios:
+        iosWorkflow === Workflow.GENERIC
+          ? DEFAULT_BARE_RUNTIME_VERSION
+          : DEFAULT_MANAGED_RUNTIME_VERSION,
+    };
 
-    Log.addNewLineIfNone();
-    if (isBare) {
-      Log.log(
-        `🧐 It seems you are on the bare workflow! Please also update your native files. You can do this by either running ${chalk.bold(
-          'eas build:configure'
-        )} or manually editing Expo.plist/AndroidManifest.xml. ${learnMore(
-          'https://expo.fyi/eas-update-config.md#native-configuration'
-        )}`
-      );
-    } else {
-      Log.log(`🎉 Your app is configured to run EAS Update!`);
+    let updatedExp: ExpoConfig;
+    try {
+      updatedExp = await configureAppJSONForEASUpdateAsync({
+        projectDir,
+        exp,
+        platform,
+        defaultRuntimeVersions,
+      });
+    } catch (e) {
+      if (androidWorkflow === Workflow.GENERIC || iosWorkflow === Workflow.GENERIC) {
+        Log.warn(
+          `You will also have to manually edit Expo.plist/AndroidManifest.xml. ${learnMore(
+            'https://expo.fyi/eas-update-config.md#native-configuration'
+          )}`
+        );
+      }
```

Comment:
Why is this in `catch`? Errors/exceptions shouldn't be used for implementing "business logic".

Reply:
Yea good point, this try/catch was unnecessary. Moved this logging here: https://github.com/expo/eas-cli/blob/073994a086efda738ec7ec4b01363223befaeefa/packages/eas-cli/src/commands/update/configure.ts#L225